File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/workdir/src/main/java/surveycrawler/TurnCounterClockwiseCommand.java
package surveycrawler;

public class TurnCounterClockwiseCommand extends CrawlerMovement {

    // testing

    @Override
    public boolean isFor(char aCharacter) {
        return aCharacter == 'g';
    }

    // executing

    @Override
    public void executeOn(SurveyCrawler aCrawler) {
        aCrawler.turnCounterClockwise();
    }

    // undoing

    @Override
    public CrawlerMovement inverse() {
        return new TurnClockwiseCommand();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/analysis/acceptance-project/src/surveycrawler/turn_counter_clockwise_command.py
from surveycrawler.command import Command


class TurnCounterClockwiseCommand(Command):

    # character

    @classmethod
    def character(cls):
        return 'g'

    # executing

    def execute_on(self, a_crawler):
        a_crawler.turn_counter_clockwise()

    def undo_on(self, a_crawler):
        a_crawler.turn_clockwise_ignoring_ground()
